src/spikeinterface/generation/hybrid_tools.py

```diff
@@ -0,0 +1,518 @@
+from __future__ import annotations
+
+import warnings
+from typing import Literal
+import numpy as np
+
+from ..core.template import Templates
+
+from ..core.generate import (
+    generate_templates,
+    generate_unit_locations,
+    generate_sorting,
+    InjectTemplatesRecording,
+    _ensure_seed,
+)
+from ..core.template_tools import get_template_extremum_channel
+from ..core.job_tools import split_job_kwargs
+from ..postprocessing.unit_localization import compute_monopolar_triangulation
+
+from .drift_tools import (
+    InjectDriftingTemplatesRecording,
+    DriftingTemplates,
+    make_linear_displacement,
+    interpolate_templates,
+    move_dense_templates,
+)
+
+
+def estimate_templates_from_recording(
+    recording, ms_before=2, ms_after=2, sorter_name="spykingcircus2", **run_sorter_kwargs
+):
+    """
+    Get templates from a recording. Internally, SpyKING CIRCUS 2 is used by default
+    with the only twist that the template matching step is not launched. Instead, a Template
+    object is returned based on the results of the clustering. Other sorters can be invoked
+    with the `sorter_name` and `run_sorter_kwargs` parameters.
+
```

Comment:
We should specified that templates is dense.

Reply:
done